src/discussions-trigger.ts

```diff
@@ -0,0 +1,175 @@
+    const discordWebhookAddress = process.env.DT_MODULE_REQ_DISCORD_WEBHOOK;
+    if (!discordWebhookAddress) throw new Error("DT_MODULE_REQ_DISCORD_WEBHOOK not set in ENV");
+
+    // https://birdie0.github.io/discord-webhooks-guide/discord_webhook.html
+    const webhook = {  content: `New DT Module requested:`, embeds: [ { title: discussion.title, url: discussion.html_url } ] };
+    await fetch(discordWebhookAddress, { method: "POST", body: JSON.stringify(webhook), headers: { "content-type": "application/json" } });
+}
+
+
+async function updateOrCreateMainComment(discussion: Discussion, message: string) {
+    const discussionComments = await getCommentsForDiscussionNumber(discussion.number);
+    const previousComment = discussionComments.find(c => c.author.login === "typescript-bot");
+    if (previousComment) {
+        await client.mutate(createMutation<any>("updateDiscussionComment" as any, { body: message, commentId: previousComment.id }));
+    } else {
+        await client.mutate(createMutation<any>("addDiscussionComment" as any, { body: message, discussionId: discussion.node_id }));
+    }
+}
+
+async function addLabel(discussion: Discussion, labelName: string, description?: string) {
+
+    const existingLabel = await getLabelByName(labelName);
+    if (existingLabel && existingLabel.name === labelName) {
+        await client.mutate(createMutation<any>("addLabelsToLabelable" as any, { labelableId: discussion.node_id, labelIds: [existingLabel.id] }));
+    } else {
+        const dtRepoID = "MDEwOlJlcG9zaXRvcnk2MDkzMzE2";
+        // https://docs.github.com/en/graphql/reference/input-objects#createlabelinput
+        const color = crypto.randomBytes(Math.ceil(6/2)).toString("hex").slice(0,6);
+
+        const newLabel = await client.mutate(createMutation("createLabel" as any, { name: labelName, repositoryId: dtRepoID, color, description })) as any;
+        const newID = newLabel.data.label.id;
+        await client.mutate(createMutation<any>("addLabelsToLabelable" as any, { labelableId: discussion.node_id, labelIds: [newID] }));
+    }
+}
+
+
+async function getLabelByName(name: string) {
+    const info = await client.query({
+        query: gql`
+      query GetLabel($name: String!) {
+        repository(name: "DefinitelyTyped", owner: "DefinitelyTyped") {
+          name
+          labels(query: $name, first: 1) {
+            nodes {
+              id
+              name
+            }
+          }
+        }
+      }      
+`,
+        variables: { name },
+        fetchPolicy: "no-cache",
+    });
+
+    return info.data.repository.labels.nodes[0] as { id: string, name: string } | undefined;
+}
+
+async function getCommentsForDiscussionNumber(number: number) {
+    const info = await client.query({
+        query: gql`
+query GetDiscussionComments($discussionNumber: Int!) {
+  repository(name: "DefinitelyTyped", owner: "DefinitelyTyped") {
+    name
+    discussion(number: $discussionNumber) {
+      comments(first: 100) {
```

Comment:
This should get them from the beginning of the thread, right?  So I think that it's better to add an `orderBy` (as in `all-open-prs-query.ts`).

Reply:
Thanks, good spot

Reply:
(didn't see the change, so leaving open.)